src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -269,17 +265,29 @@ applyCoinbase v logger dbEnv (Miner mid mks) reward@(ParsedDecimal d) pd parentH
     forkTime = vuln797FixDate v
     fork1_3InEffect = blockTime >= forkTime
     throwCritical = fork1_3InEffect || enfCBFailure
-    blockTime = blockTimeOf pd
-
     ec = mkExecutionConfig
       [ FlagDisableModuleInstall
       , FlagDisableHistoryInTransactionalMode ]
     tenv = TransactionEnv Transactional dbEnv logger pd noSPVSupport
            Nothing 0.0 rk 0 ec
     txst = TransactionState mc mempty 0 Nothing (_geGasModel freeGasEnv)
     initState = setModuleCache mc $ initCapabilities [magic_COINBASE]
-    chash = Pact.Hash (sshow $ _blockHash parentHeader)
     rk = RequestKey chash
+    bh = fromIntegral $ _pdBlockHeight pd
+    blockTime = blockTimeOf pd
+        -- NOTE it should hold that @blockTime == _blockCreationTime parentHeader@, but
+        -- in some unit test runs that is not the case. That is fine because coinbase
+        -- is a special case
```

Comment:
I figured this out btw. Stuart's test for the forking change validations makes use of blocktime in the publidata _only_, without using a parent header to generate the data as I do in `PactService.runCoinbase`. This leads to the error message when we remove this code and only rely on the parent header for time data. Technically, this should be fine, since its generated via `mkPublicData`, which makes use of the parent header in service state. I'm fine with not poking it if we don't have to, removing it when we move to blockheights

Reply:
Agree, that we can leave it like this for now. I just put the note in there to not get confused again in the future.

Reply:
see my review https://github.com/kadena-io/chainweb-node/pull/902#pullrequestreview-362793443 , we should fix this now/standardize on basing all logic on injected `parentHeader` and remove comments.

Reply:
Some of these code lines are going to disappear in one of the follow up PRs. Let's post pone the discussion about current vs. parent header to that PR.